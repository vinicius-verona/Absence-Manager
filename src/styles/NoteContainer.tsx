import styled from "styled-components";

export const TextArea = styled.textarea`
  wrap: soft;
  resize: none;
  rows: 10;
  readonly: true;
  border-radius: 10px;
`;

export const NoteWrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`;
