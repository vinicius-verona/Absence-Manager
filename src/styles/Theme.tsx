// eslint-disable-next-line import/no-anonymous-default-export
export default {
    palettes: {
        primary: "rgba(255, 148, 25, 1)",
        lightPrimary: "rgba(255, 148, 25, 0.5)",
        light: "#DFDFDF",
        darkerLight: "#B5B5B5",
        bg_dark: "rgba(0, 0, 0, 0.5)",
    },
};
