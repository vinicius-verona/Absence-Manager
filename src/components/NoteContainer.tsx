import { NoteWrapper, TextArea } from "../styles/NoteContainer";

type NoteContent = {
  label: string;
  content: string;
};

/**
 * Creates a text-area to display notes.
 * @param props - the label to be displayed and the content
 * * props = {`label`: string, `content`: string}
 */
export default function NoteContainer(props: NoteContent) {
  return (
    <NoteWrapper>
      <label>
        <strong>{props.label}</strong>
      </label>
      <TextArea readOnly={true} value={props.content} />
    </NoteWrapper>
  );
}
