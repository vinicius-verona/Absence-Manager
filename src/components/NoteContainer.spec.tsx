import { render, screen } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import renderer from "react-test-renderer";
import Theme from "../styles/Theme";
import NoteContainer from "./NoteContainer";

describe("Render absence' note container", () => {
  it("Should match note container snapshot", () => {
    const container = renderer
      .create(
        <ThemeProvider theme={Theme}>
          <NoteContainer label={"Member:"} content={""} />
        </ThemeProvider>
      )
      .toJSON();

    expect(container).toMatchSnapshot();
  });

  it("Should render label 'Member:'", () => {
    const { getByText } = render(
      <ThemeProvider theme={Theme}>
        <NoteContainer label={"Member:"} content={""} />
      </ThemeProvider>
    );

    expect(getByText("Member:")).toBeInTheDocument();
  });

  it("Should render note 'Member Test Note'", () => {
    const { getByText } = render(
      <ThemeProvider theme={Theme}>
        <NoteContainer label={""} content={"Member Test Note"} />
      </ThemeProvider>
    );

    expect(getByText("Member Test Note")).toBeInTheDocument();
  });

  it("Should render note label by parameter", async () => {
    const { getByText, rerender } = render(
      <ThemeProvider theme={Theme}>
        <NoteContainer label={"Member:"} content={""} />
      </ThemeProvider>
    );

    expect(getByText("Member:")).toBeInTheDocument();

    rerender(
      <ThemeProvider theme={Theme}>
        <NoteContainer label={"Admitter:"} content={""} />
      </ThemeProvider>
    );

    expect(screen.queryByText("Member:")).not.toBeInTheDocument();
    expect(screen.getByText("Admitter:")).toBeInTheDocument();
  });

  it("Should render note content by parameter", async () => {
    const { getByText, rerender } = render(
      <ThemeProvider theme={Theme}>
        <NoteContainer label={""} content={"Member Test Note"} />
      </ThemeProvider>
    );

    expect(getByText("Member Test Note")).toBeInTheDocument();

    rerender(
      <ThemeProvider theme={Theme}>
        <NoteContainer label={""} content={"Admitter Test Note"} />
      </ThemeProvider>
    );

    expect(screen.queryByText("Member Test Note")).not.toBeInTheDocument();
    expect(screen.getByText("Admitter Test Note")).toBeInTheDocument();
  });
});
