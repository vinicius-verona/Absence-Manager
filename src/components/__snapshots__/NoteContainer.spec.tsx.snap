// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render absence' note container Should match note container snapshot 1`] = `
<div
  className="sc-gsDKAQ fIdEjT"
>
  <label>
    <strong>
      Member:
    </strong>
  </label>
  <textarea
    className="sc-bdvvtL cReOru"
    readOnly={true}
    value=""
  />
</div>
`;
